[Frontend] Fix BarChart and MetricsCategoryBarCharts new lint errors in `common` repo

**BarChart.tsx**

```
  158:21  error  Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component and pass data as props. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
```


**MetricsCategoryBarChart.tsx**

```
  126:23  error  Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component and pass data as props. If you want to allow component creation in props, set allowAsProps option to true  react/no-unstable-nested-components
```

The errors are referring specifically to us defining an anonymous function within the `tick` prop. We can define this before the render and pass in the function reference to the `tick` prop.

```
              <YAxis
                tick={(props: TickProps) => (
                  <CustomYAxisTick
                    y={props.y}
                    payload={props.payload}
                    percentageView={false}
                    styles={tickStyle}
                  />
                )}
                ...
```

--- a/common/components/DataViz/BarChart.tsx
+++ b/common/components/DataViz/BarChart.tsx
@@ -114,6 +114,15 @@ const ResponsiveBarChart = forwardRef<never, ResponsiveBarChartProps>(
           height: 500,
         };
 
+    const renderCustomYAxisTick = (props: TickProps) => (
+      <CustomYAxisTick
+        y={props.y}
+        payload={props.payload}
+        percentageView={!!percentageView}
+        styles={tickStyle}
+      />
+    );
+
     return (
       <ChartContainer>
         <ResponsiveContainer {...responsiveContainerProps}>
@@ -157,16 +166,7 @@ const ResponsiveBarChart = forwardRef<never, ResponsiveBarChartProps>(
             />
             <YAxis
               allowDecimals={percentageView}
-              // TODO(#803): Fix lint error and remove the `eslint-disable-next-line`
-              // eslint-disable-next-line react/no-unstable-nested-components
-              tick={(props: TickProps) => (
-                <CustomYAxisTick
-                  y={props.y}
-                  payload={props.payload}
-                  percentageView={!!percentageView}
-                  styles={tickStyle}
-                />
-              )}
+              tick={renderCustomYAxisTick}
               tickLine={false}
               tickCount={percentageView ? 5 : 12}
               domain={percentageView ? ["dataMin", "dataMax"] : undefined}

--- a/common/components/DataViz/MetricsCategoryBarChart.tsx
+++ b/common/components/DataViz/MetricsCategoryBarChart.tsx
@@ -102,6 +102,18 @@ const MetricsCategoryBarChart = forwardRef<never, ResponsiveBarChartProps>(
       );
       return barDefinitions;
     };
+
+    const renderCustomYAxisTick = (props: TickProps) => (
+      <CustomYAxisTick
+        y={props.y}
+        payload={props.payload}
+        percentageView={false}
+        styles={tickStyle}
+        metric={metric}
+        rightAligned
+      />
+    );
+
     return (
       <>
         <ResponsiveContainer width={width} height="100%">
@@ -142,17 +154,7 @@ const MetricsCategoryBarChart = forwardRef<never, ResponsiveBarChartProps>(
             />
             {data.length !== 0 && (
               <YAxis
-                // eslint-disable-next-line react/no-unstable-nested-components
-                tick={(props: TickProps) => (
-                  <CustomYAxisTick
-                    y={props.y}
-                    payload={props.payload}
-                    percentageView={false}
-                    styles={tickStyle}
-                    metric={metric}
-                    rightAligned
-                  />
-                )}
+                tick={renderCustomYAxisTick}
                 tickLine={false}
                 tickCount={5}
                 axisLine={false}

